Core/HttpHandlerProxy.cs

```diff
@@ -0,0 +1,151 @@
+                            : networkCredential.UserName + ":" + networkCredential.Password;
+
+                        string base64AuthString = Convert.ToBase64String(Encoding.UTF8.GetBytes(authString));
+
+                        request.Headers.Authorization = new AuthenticationHeaderValue("Basic", base64AuthString);
+                    }
+                    break;
+
+                case null:
+                    break;
+
+                default:
+                    throw new AuthenticationException($"Credentials of type '{Credentials.GetType().Name}' is not supported");
+            }
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var cookieHeader = CookieContainer?.GetCookieHeader(request.RequestUri);
+            if (!string.IsNullOrEmpty(cookieHeader))
+            {
+                request.Headers.Add(HeaderNames.Cookie, cookieHeader);
+            }
+
+            Authorize(request);
+
+            HttpResponseMessage response = null;
+            if (null != InnerHandler)
+            {
+                response = await base.SendAsync(request, cancellationToken);
+            }
+            else if (null != httpMessageHandler)
+            {
+                var delegatingHandler = new SimpleHttpMessageHandlerProxy(httpMessageHandler);
+                response = await delegatingHandler.SendImplAsync(request, cancellationToken);
+
+                // SB: special case for Negotiate authentication, we can't reuse named HttpMessageHandler,
+                // since this type of auth requires credentials to be set on handler level, after that
+                // handler can't be reused with another credentials
+                if (response.StatusCode == HttpStatusCode.Unauthorized && Credentials is NetworkCredential && response.Headers?.WwwAuthenticate?.Count > 0)
+                {
+                    delegatingHandler.Dispose();
+                    delegatingHandler = null;
+
+                    var handler = HttpClientExtensions.CreateHttpClientHandler(httpClientName);
+                    handler.Credentials = Credentials;
+
+                    InnerHandler = handler;
+
+                    response = await base.SendAsync(request, cancellationToken);
+                }
+                else
+                {
+                    InnerHandler = delegatingHandler;
+                }
+            }
+            else if (null != httpClientHandler)
+            {
+                httpClientHandler.Credentials = Credentials;
+                InnerHandler = httpClientHandler;
+
+                response = await base.SendAsync(request, cancellationToken);
+            }
+
+            if (response.Headers.TryGetValues(HeaderNames.SetCookie, out var values))
+            {
+                values.ForEach(c => CookieContainer?.SetCookies(request.RequestUri, c));

```

Comment:
Note, there are some specifics https://github.com/dotnet/runtime/issues/26776

Reply:
in case of EWS I have never seen only host name in URL, it is always EWS/Exchange.asmx